actionpack/lib/action_dispatch/middleware/ssl.rb

```diff
@@ -123,20 +123,12 @@ def flag_cookies_as_secure!(headers)
       end
 
       def redirect_to_https(request)
-        [ @redirect.fetch(:status, redirection_status(request)),
+        [ @redirect.fetch(:status, Rack::Utils::SYMBOL_TO_STATUS_CODE[:permanent_redirect]),
           { "Content-Type" => "text/html",
             "Location" => https_location_for(request) },
           (@redirect[:body] || []) ]
       end
 
-      def redirection_status(request)
-        if request.get? || request.head?
-          301 # Issue a permanent redirect via a GET request.
```

Comment:
There is no need for us to use `301` anymore according to [RFC 7538](https://tools.ietf.org/html/rfc7538#section-3)

```
   A 308 response is cacheable by default; i.e., unless otherwise
   indicated by the method definition or explicit cache controls (see
   [RFC7234], Section 4.2.2).

      Note: This status code is similar to 301 (Moved Permanently)
      ([RFC7231], Section 6.4.2), except that it does not allow changing
      the request method from POST to GET.
```

Reply:
Get the idea, but I'm wondering all the apps having behaviors or test expecting 301 and be changed to 308 without previous notification...

Reply:
I think this will have to be a breaking change for Rails 6.1 since apps with tests expecting `307` would break as well.

Reply:
I agree it is a breaking change, however, since this is a library\gem code, that is being checked here, only integration tests that will test that the ssl_force option is activated for their application will be influenced, and the fix is pretty simple